src/Octoshift/AdoApi.cs

```diff
@@ -27,7 +27,7 @@ public AdoApi(AdoClient client, string adoServerUrl, OctoLogger log)
 
         public virtual async Task<string> GetOrgOwner(string org)
         {
-            var url = $"{_adoBaseUrl}/{org}/_apis/Contribution/HierarchyQuery?api-version=5.0-preview.1";
+            var url = $"{_adoBaseUrl}/{org.UrlEncode()}/_apis/Contribution/HierarchyQuery?api-version=5.0-preview.1";
```

Comment:
1. `UrlEncode` is not what we want here (I know it is confusing), what we actually want is url escaping (or percent encoding if you will). The reason is `HttpUtility.UrlEncode()` acts a bit differently and encodes the spaces to `+` rather than percent encode to `%20`. The recommended way is to use [Uri.EscapeUriString()](https://learn.microsoft.com/en-us/dotnet/api/system.uri.escapeuristring?view=net-7.0) with `Uri.EscapeDataString()` (for escaping query paramerters).
2. In most cases with can get away with just passing the entire url into the `Uri.EscapeUriString()` except in cases where we construct the query string and it needs to be escaped in which the `Uri.EscapeDataString()` would be the recommended option (because it escapes the reserved chars too). Here is a good [SO thread](https://stackoverflow.com/questions/4396598/whats-the-difference-between-escapeuristring-and-escapedatastring) about `EscapeUriString` vs `EscapeDataString`.

Reply:
That SO thread explains it well, I'll change it to EscapeDataString()

Reply:
Done. Changed it to EscapeDataString everywhere I had UrlEncode